src/CarbonAware.LocationSources/src/Configuration/LocationDataSource.cs

```diff
@@ -0,0 +1,56 @@
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace CarbonAware.LocationSources.Configuration;
+
+public class LocationDataSource
+{
+
+    private const string DefaultAzureLocationDataFile = "azure-regions.json";
```

Comment:
> We should consider not hard-coding Azure here. I would prefer the default to just be everything in the `location-sources/json` (that would be managed elsewhere obviously)

Good point. Should we rename the file too? I don't have a strong preference to keep this `azure-regions.json` file named with `azure`

Reply:
@juzuluag Do we need a default file at all?  That seems like something which should be specified on instantiation.

Reply:
We need one to have 'backward' compatibility. Not sure whether this should be part of an `appsettings.json` or not, since if the dev decides to cleanup the json file, then something breaks. But I'm open too for pros/cons of having it as part of the appsettings

Reply:
What if we rewrote `LoadLocationJsonFileAsync` to manage defaults this way:

```csharp
private async Task LoadLocationJsonFileAsync() {
  var locationFiles = _configuration.LocationSourceFiles.Any() ? _configuration.LocationSourceFiles : Directory.GetFiles(LocationSourceFile.FilePath).Select( fileName => new LocationSourceFile(fileName) );

  foreach (var locationFile in locationFiles)
  {
    using Stream stream = GetStreamFromFileLocation(locationFile );
    var namedGeopositions = await JsonSerializer.DeserializeAsync<List<NamedGeoposition>>(stream, _options);
    foreach (var namedGeoposition in namedGeopositions!) 
    {
        _namedGeopositions.Add(BuildKeyFromRegion(locationFile , namedGeoposition), namedGeoposition);
    }
  }
}
```

Then any default files can be included in the `data/location-sources` and they will all get used if the operator doesn't specify.  In that context having a list of azure regions feels appropriate because it's easy enough for others to drop major cloud provider files in as defaults without touching the code.

Reply:
> What if we rewrote `LoadLocationJsonFileAsync` to manage defaults this way:
> 
> ```cs
> private async Task LoadLocationJsonFileAsync() {
>   var locationFiles = _configuration.LocationSourceFiles.Any() ? _configuration.LocationSourceFiles : Directory.GetFiles(LocationSourceFile.FilePath).Select( fileName => new LocationSourceFile(fileName) );
> 
>   foreach (var locationFile in locationFiles)
>   {
>     using Stream stream = GetStreamFromFileLocation(locationFile );
>     var namedGeopositions = await JsonSerializer.DeserializeAsync<List<NamedGeoposition>>(stream, _options);
>     foreach (var namedGeoposition in namedGeopositions!) 
>     {
>         _namedGeopositions.Add(BuildKeyFromRegion(locationFile , namedGeoposition), namedGeoposition);
>     }
>   }
> }
> ```
> 
> Then any default files can be included in the `data/location-sources` and they will all get used if the operator doesn't specify. In that context having a list of azure regions feels appropriate because it's easy enough for others to drop major cloud provider files in as defaults without touching the code.

ok. I like the idea of 'discovering' location files. This would avoid to know more about them. The only **warning**  here is name collision between files with the prefix. Unless we don't care, there are going to be cases where the region name could be the same between different json files.
I'll impl this proposal.